### PR TITLE
Feature/back compatibility

### PR DESCRIPTION
**Pull request by dbmeneses**

### CODE REVIEW

File: sonar-runner-batch/src/main/java/org/sonar/runner/batch/BatchIsolatedLauncher.java [hunk 1]
```diff
@@ -65,12 +66,17 @@ public void syncProject(String projectKey) {
     batch.syncProject(projectKey);
   }
 
-  Batch createBatch(Properties properties, @Nullable final org.sonar.runner.batch.LogOutput logOutput) {
+  Batch createBatch(Properties properties, @Nullable final org.sonar.runner.batch.LogOutput logOutput, @Nullable List<Object> extensions) {
+    // initLogging(properties);
```
Comment (SonarTech) [inline sonar-runner-batch/src/main/java/org/sonar/runner/batch/BatchIsolatedLauncher.java]: MAJOR This block of commented-out lines of code should be removed. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

File: sonar-runner-api/src/main/java/org/sonar/runner/api/LoggerAdapter.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+  @Override
+  public void info(String msg) {
+    logOutput.log(msg, LogOutput.Level.INFO);
+  }
+
+  @Override
+  public void error(String msg, Throwable t) {
+    StringWriter errors = new StringWriter();
+    t.printStackTrace(new PrintWriter(errors));
```
Comment (SonarTech) [inline sonar-runner-api/src/main/java/org/sonar/runner/api/LoggerAdapter.java]: CRITICAL Use a logger to log this exception. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1148)

File: sonar-runner-api/src/main/java/org/sonar/runner/impl/SimulatedLauncher.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+import java.io.FileOutputStream;
+import java.io.OutputStream;
+import java.util.List;
+import java.util.Properties;
+
+import org.sonar.runner.batch.IsolatedLauncher;
+
+public class SimulatedLauncher implements IsolatedLauncher {
+  private final String version;
+  private final Logger logger;
```
Comment (SonarTech) [inline sonar-runner-api/src/main/java/org/sonar/runner/impl/SimulatedLauncher.java]: MAJOR Make the "logger" logger private static final and rename it to comply with the format "LOG(?:GER)?". [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1312)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues:
* CRITICAL 1 critical
* MAJOR 2 major

Watch the comments in this conversation to review them.

**Comment by henryju** [comment]
LGTM

**Comment by dbmeneses** [comment]
merged

**Comment by dbolkensteyn** [comment]
FYI @dbmeneses , if you rebase your PR, force push to the `feature/back_compatibility` branch, and then merge it into the master, GitHub will automatically detect the merge and show a nicer "Merged" icon instead of the red "Close" :)

**Comment by dbmeneses** [comment]
I know, in this case I did a simple correction before merging and didn't
want to launch the on-push jobs on the branch. Anyway, I delete the branch
immediately after (and to delete, I need to close it).
Next time I'll do it and then cancel the jobs.

On 28 August 2015 at 08:46, Dinesh Bolkensteyn <notifications@github.com>
wrote:

> FYI @dbmeneses <https://github.com/dbmeneses> , if you rebase your PR,
> force push to the feature/back_compatibility branch, and then merge it
> into the master, GitHub will automatically detect the merge and show a
> nicer "Merged" icon instead of the red "Close" :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SonarSource/sonar-runner/pull/17#issuecomment-135655372>
> .
>



-- 
*Duarte MENESES | **SonarSource*
http://sonarsource.com

**Pull request closed without merging**